src/droplet.F

```diff
@@ -2270,6 +2419,196 @@ subroutine ie_vrt_nd(diffnorm,tempr,tempt,rt_output,dt_nondim,taup_scale,prsval,
+      real :: u10
+
+      real :: xrange,yrange,zrange
+      real :: salinity
+      
+      real :: c1,c2,c3,u14,cdn10,a,rand
+      real :: a1,a2,r80,r0,dh
+      real :: dFsdr80(100),dFmsdr0(100),dr80_dr0
+      integer :: iter,nbin,num_create,j
+
+      !implementing the Andreas 1998 sea spray generation function
+      !Set the parameters of the two lognormals:
+      real :: rmin,rmax,rmin10,rmax10
+
+      rmin10 = log10(2e-06)
+      rmax10 = log10(500e-06)
+      dh = 0.0242
+
+  !   ===== update x-axis for each bin =====
+      binsdata(1) = rmin10
+      do i = 1,99
+        binsdata(i+1)= dh+binsdata(i)
+      end do
+
+      binsdata = 1e6*(10**binsdata)
+  
+      ! Now figure out how many particles are produced in each of
+      ! those bins. This is a function of u10
+      u10 = min(s10_xym,50.0)
+
+      c1 = 10*smithssgf(u10,10.0,0.4)
+      c2 = c1*(37.5**1.8)
+      c3 = c2*(100**5.2)
+      dFssum(1) = 0
+      do i = 1,100
+        r0 = binsdata(i)
+        ! calculate radius at 80%RH
+        r80 = 0.518*r0**0.976;
+ 
+     ! now apply these to Eqs. 3.5 in Andreas 1998
+        if (r80 .lt. 10) then
+          dFsdr80(i) =  smithssgf(u10,r80,0.4)
+        elseif ((r80 .ge. 10) .and. (r80 .lt. 37.5)) then
+          dFsdr80(i) = c1/r80
+        elseif ((r80 .ge. 37.5) .and. (r80 .lt. 100)) then
+          dFSdr80(i) = c2*r80**-2.8
+        elseif ((r80 .ge. 100) .and. (r80 .lt. 250)) then
+          dFsdr80(i) = c3*r80**-8
+        endif
+ 
+       ! apply eq. 3.8 from Andreas 1998
+       dFmsdr0(i) = 3.5*dFsdr80(i)*0.506*r0**-0.024
+ 
+       if (i==1) then
+         dFssum(i+1) = dFmsdr0(i)
+       elseif (i .gt. 1) then
+         dFssum(i+1) = dFssum(i) + dFmsdr0(i)
+       endif
+  
+      end do
+
+      totdrops = dFssum(101)    !totdrops at this point is # per unit area per unit time
+
+      !Define these here
+      xrange = xf(ni+1)-xf(1)
+      yrange = yf(nj+1)-yf(1)
+      zrange = 8.0
+
+#ifdef MPI
+      my_reintro = floor(xrange*yrange*dt*totdrops/(numprocs*drop_mult))    !Number of droplets to be produced by this MPI task
+#else
+      my_reintro = floor(xrange*yrange*dt*totdrops/drop_mult)      !Number of droplets to be produced by this MPI task
+#endif
+
+      my_reintro = 10
+
+   if (myid==0) write(*,'(a15,i,6e15.6)')  'DROP INJECTION:',my_reintro,drop_mult,u10,totdrops,xrange,yrange,dt
+
+      if (nparcelsActive+my_reintro .lt. nparcelsActiveMax) then   !Only add particles to pdata if we're still under nparcelsActiveMax
+
+      do np=nparcelsActive+1,nparcelsActive+my_reintro
+
+         rand = getRandomReal(randomNumbers)
+         a = totdrops*getRandomReal(randomNumbers)
+
+         do i = 1,100
+            if ((a .gt. dFssum(i)) .and. (a .le. dFssum(i+1))) then
+               rand = getRandomReal(randomNumbers)
+               rad_init = binsdata(i) + rand*(binsdata(i+1)-binsdata(i))
+            endif
+         enddo
```

Comment:
Can you add some comments about what this loop is doing here?

Reply:
I have added some comments. Let me know if they don't explain well